ee/katc/sqlite.go

```diff
@@ -0,0 +1,84 @@
+package katc
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log/slog"
+	"path/filepath"
+
+	_ "modernc.org/sqlite"
+)
+
+// sqliteData is the dataFunc for sqlite KATC tables
+func sqliteData(ctx context.Context, pathPattern string, query string, columns []string, slogger *slog.Logger) ([]map[string][]byte, error) {
+	sqliteDbs, err := filepath.Glob(pathPattern)
+	if err != nil {
+		return nil, fmt.Errorf("globbing for files with pattern %s: %w", pathPattern, err)
+	}
+
+	results := make([]map[string][]byte, 0)
+	for _, sqliteDb := range sqliteDbs {
+		resultsFromDb, err := querySqliteDb(ctx, sqliteDb, query, columns, slogger)
+		if err != nil {
+			return nil, fmt.Errorf("querying %s: %w", sqliteDb, err)
+		}
+		results = append(results, resultsFromDb...)
+	}
+
+	return results, nil
+}
+
+// querySqliteDb queries the database at the given path, returning rows of results
+func querySqliteDb(ctx context.Context, path string, query string, columns []string, slogger *slog.Logger) ([]map[string][]byte, error) {
+	conn, err := sql.Open("sqlite", path)
+	if err != nil {
+		return nil, fmt.Errorf("opening sqlite db: %w", err)
+	}
+	defer func() {
+		if err := conn.Close(); err != nil {
+			slogger.Log(ctx, slog.LevelWarn,
+				"closing sqlite db after query",
+				"err", err,
+			)
+		}
+	}()
+
+	rows, err := conn.QueryContext(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("running query: %w", err)
+	}
+	defer func() {
+		if err := rows.Close(); err != nil {
+			slogger.Log(ctx, slog.LevelWarn,
+				"closing rows after scanning results",
+				"err", err,
+			)
+		}
+	}()
+
+	results := make([]map[string][]byte, 0)
```

Comment:
Does this drop `path` ?

Reply:
It does -- should we include it?

Reply:
Yes. A common ATC table is paths like `/Users/%/.config/blah.sql` so you need the path to differentiate between users.

Reply:
Path is now included